packages/react/src/NavList/NavList.tsx

```diff
@@ -288,6 +290,54 @@ const Group: React.FC<NavListGroupProps> = ({title, children, sx: sxProp = defau
   )
 }
 
+export type NavListShowMoreItemProps = {
+  children: React.ReactNode
+  label: string
+} & SxProp
+
+const ItemWithinGroup = React.createContext<{groupId: string} | null>(null)
+
+const ShowMoreItem = React.forwardRef<HTMLButtonElement, NavListShowMoreItemProps>(
+  ({label, children, ...props}, forwardedRef) => {
+    const [expanded, setExpanded] = React.useState(false)
+    const targetFocused = React.useRef(false)
+
+    const groupId = useId()
+
+    React.useEffect(() => {
+      if (expanded && !targetFocused.current) {
+        const focusTarget: HTMLAnchorElement | null = document.querySelector(`[data-show-more-group-id="${groupId}"]`)
+
+        if (focusTarget) {
+          focusTarget.focus()
+          targetFocused.current = true
+        }
+      }
+    }, [expanded, groupId])
+
+    return !expanded ? (
+      <Box as="li" sx={{listStyle: 'none'}}>
+        <ActionList.Item
+          as="button"
+          aria-expanded="false"
+          ref={forwardedRef}
+          onClick={() => {
+            setExpanded(true)
+          }}
+          {...props}
+        >
+          {label}
+          <ActionList.TrailingVisual>
+            <PlusIcon />
+          </ActionList.TrailingVisual>
+        </ActionList.Item>
+      </Box>
+    ) : (
+      <ItemWithinGroup.Provider value={{groupId}}>{children}</ItemWithinGroup.Provider>
```

Comment:
Could we memoize this value since we're passing it to a context provider? Would help with unnecessary re-renders since this is an object defined inline.